test/Libraries/RevitNodesTests/Elements/ElementTests.cs

```diff
@@ -436,5 +437,35 @@ private static void AssertElementsAreJoined(Element element, Element otherElemen
         }
 
         #endregion
+
+        [Test]
+        [TestModel(@".\Element\elementJoin.rvt")]
+        public void CanSuccessfullyJoinTwoIntersectingElements()
+        {
+            Document doc = DocumentManager.Instance.CurrentDBDocument;
+            var primaryBeam = ElementSelector.ByElementId(208422, true);
+            var nonIntersectingBeam = ElementSelector.ByElementId(209681, true);
+            var joinedBeam = ElementSelector.ByElementId(208572, true);
+            var notJoinedWall = ElementSelector.ByElementId(207960, true);
+            var notJoinedFloor = ElementSelector.ByElementId(208259, true);
+
+            System.Type nonIntersectingTestExpectedExceptionType = typeof(System.NullReferenceException);
+            string nonIntersectingTestExpectedExceptionString = "Elements are not intersecting";
+
+            List<int> joinedTestExpectedOutcome = new List<int> { 208422, 208572 };
+            List<int> notJoinedTestExpectedOutcome = new List<int> { 207960, 208259 };
+
+            var alreadyJoinedOutcome = primaryBeam.JoinGeometry(joinedBeam).Select(elem => elem.Id).ToList();
+            var notJoinedIntersectingOutcome = notJoinedWall.JoinGeometry(notJoinedFloor).Select(elem => elem.Id).ToList();
+
+            Assert.AreEqual(joinedTestExpectedOutcome, alreadyJoinedOutcome);
+            Assert.AreEqual(notJoinedTestExpectedOutcome, notJoinedIntersectingOutcome);
+
+            // Non intersecting elements should throw a NullReferenceException
+            // with the messages Elements are not intersecting
+            var ex = Assert.Throws<NullReferenceException>(() => primaryBeam.JoinGeometry(nonIntersectingBeam)); 
+            Assert.That(ex.Message, Is.EqualTo(nonIntersectingTestExpectedExceptionString));
```

Comment:
`Assert.AreEqual` should do the trick, don't think this complex syntax is needed.